lib/rules/no-assignment-of-untracked-properties-used-in-tracking-contexts.js

```diff
@@ -74,29 +158,64 @@ module.exports = {
       return {};
     }
 
+    const options = context.options[0] || { extraMacros: [] };
+
+    const macroConfigurations = [...DEFAULT_MACRO_CONFIGURATIONS, ...options.extraMacros];
+
+    const macrosByPath = new Map();
```

Comment:
Can we extract a helper function to create these maps instead of using `forEach`?

```js
const macrosByPath = createMacrosByPathMap(macroConfigurations);
const macrosByIndexPath = createMacrosByIndexPathMap(macroConfigurations);
```

That would make it more clear what this code does (could include jsdoc comments with the functions), encapsulate the complex logic, and make it possible to unit test it.